subsys/bluetooth/controller/power_clock_driver.c

```diff
@@ -10,20 +10,32 @@
 #include <kernel_includes.h>
 #include <clock_control.h>
 #include <ble_controller_soc.h>
+#include "multithreading_lock.h"
 
 static int hf_clock_start(struct device *dev, clock_control_subsys_t sub_system)
 {
 	ARG_UNUSED(dev);
 
-	if (ble_controller_hf_clock_request(NULL) != 0) {
+	int32_t errcode = MULTITHREADING_LOCK_ACQUIRE();
+	if (errcode == 0) {
+		errcode = ble_controller_hf_clock_request(NULL);
+		MULTITHREADING_LOCK_RELEASE();
+	}
+	if (errcode != 0) {
 		return -EFAULT;
 	}
 
 	bool blocking = POINTER_TO_UINT(sub_system);
 	if (blocking) {
 		bool is_running = false;
 		while (!is_running) {
-			if (ble_controller_hf_clock_is_running(&is_running) != 0) {
+			errcode = MULTITHREADING_LOCK_ACQUIRE();
+			if (errcode == 0) {
+				errcode = ble_controller_hf_clock_is_running(
+					&is_running);
+				MULTITHREADING_LOCK_RELEASE();
+			}
+			if (errcode != 0) {
 				return -EFAULT;
```

Comment:
I see this several places in this file.

Reply:
I am just maintaining error codes. You can have a look at the base version. API error code is ignored and some specific error code is returned.